core/templates/dev/head/domain/topic/TopicUpdateService.js

```diff
@@ -230,6 +228,61 @@ oppia.factory('TopicUpdateService', [
           // Get the current change list.
           var currentChangeList = UndoRedoService.getChangeList();
           var indicesToDelete = [];
+          var oldSubtopicIdSkillIdMap = {};
```

Comment:
It might be better to initialize this with the subtopic ID / skill ID mapping at the outset. That way you don't have to have an additional if/else check that's based only on your local record-keeping.

Reply:
Is that necessary? Since, I am using that local record keeping as a part of the entire logic to mark the beginning index of a move operation related to a skill as the loop iterates through the changelist. Initialising at the beginning would require another loop.

Reply:
Done, yeah, this does seem like a simpler solution. Thanks!